framework/include/problems/FEProblemBase.h

```diff
@@ -766,6 +766,37 @@ class FEProblemBase : public SubProblem, public Restartable
   virtual void swapBackMaterialsFace(THREAD_ID tid);
   virtual void swapBackMaterialsNeighbor(THREAD_ID tid);
 
+  /**
+   * Method for creating an adding an object to the warehouse.
+   *
+   * @tparam T The base object type (registered in the Factory)
+   * @param type String type of the object (registered in the Factory)
+   * @param name Name for the object to be created
+   * @param parameters InputParameters for the object
+   * @param threaded Whether or not to create n_threads copies of the object
+   * @return A vector of shared_ptrs to the added objects
+   */
+  template <typename T>
+  std::vector<std::shared_ptr<T>> addObject(const std::string & type,
+                                            const std::string & name,
+                                            InputParameters & parameters,
+                                            const bool threaded = true)
+  {
+    // Add the _subproblem and _sys parameters depending on use_displaced_mesh
+    addObjectParamsHelper(parameters);
+
+    const auto n_threads = threaded ? libMesh::n_threads() : 1;
+    std::vector<std::shared_ptr<T>> objects(n_threads);
+    for (THREAD_ID tid = 0; tid < n_threads; ++tid)
+    {
+      std::shared_ptr<T> obj = _factory.create<T>(type, name, parameters, tid);
+      theWarehouse().add(obj);
+      objects[tid] = std::move(obj);
+    }
+
+    return objects;
+  }
```

Comment:
I would prefer that this body be outside the class definition. Takes up too much space for a developer perusing through APIs

Reply:
I'm not sure how I would go about doing that with this being a templated method.

Reply:
Oh - disregard. I follow. Sounds good.